go/obscuronode/enclave/core/crypto.go

```diff
@@ -1,22 +1,43 @@
 package core
 
 import (
+	"crypto/cipher"
+
 	"github.com/ethereum/go-ethereum/rlp"
 	"github.com/obscuronet/obscuro-playground/go/log"
 	"github.com/obscuronet/obscuro-playground/go/obscuronode/nodecommon"
 )
 
+// TODO - This fixed nonce is insecure, and should be removed alongside the fixed rollup encryption key.
+const rollupCipherNonce = "000000000000"
+
 // todo - this should become an elaborate data structure
 type SharedEnclaveSecret []byte
 
-func DecryptTransactions(txs nodecommon.EncryptedTransactions) L2Txs {
-	t := make([]nodecommon.L2Tx, 0)
-	for _, tx := range txs {
-		t = append(t, DecryptTx(tx))
+func EncryptTransactions(transactions L2Txs, rollupCipher cipher.AEAD) nodecommon.EncryptedTransactions {
+	encodedTxs, err := rlp.EncodeToBytes(transactions)
+	if err != nil {
+		log.Panic("could not encrypt L2 transaction. Cause: %s", err)
 	}
-	return t
+
+	return rollupCipher.Seal(nil, []byte(rollupCipherNonce), encodedTxs, nil)
+}
+
+func DecryptTransactions(encryptedTxs nodecommon.EncryptedTransactions, rollupCipher cipher.AEAD) L2Txs {
+	encodedTxs, err := rollupCipher.Open(nil, []byte(rollupCipherNonce), encryptedTxs, nil)
+	if err != nil {
+		log.Panic("could not decrypt encrypted L2 transactions. Cause: %s", err)
+	}
+
+	txs := L2Txs{}
+	if err := rlp.DecodeBytes(encodedTxs, &txs); err != nil {
+		log.Panic("could not decode encoded L2 transactions. Cause: %s", err)
+	}
+
+	return txs
 }
 
+// TODO - Perform real decryption here, and not just RLP decoding.
 func DecryptTx(tx nodecommon.EncryptedTx) nodecommon.L2Tx {
```

Comment:
I think we should somehow differentiate here between the "Encrpt/Decrypt Tx" and "Encrpt/Decrypt Transaction*s*" functions, because they are very different. 
The former are for communication with the users, and use the assymetric key of the enclave.
The latter are for the rollups and use the symmetric key(s)